experimental/segmenter/src/rule_segmenter.rs

```diff
@@ -9,6 +9,18 @@ use crate::symbols::*;
 use core::str::CharIndices;
 use utf8_iter::Utf8CharIndices;
 
+/// The category tag that is returned by rule_status.
+#[non_exhaustive]
+#[derive(PartialEq, Debug)]
+pub enum RuleStatusType {
+    /// No category tag
+    None = 0,
```

Comment:
I think you need `#[repr(u8)]`, as you cast to `u8` in datagen.
Consider storing the enum directly in the `ZeroVec`, so you don't have to do the conversions.

Reply:
you don't need to set the repr for the cast to work

Reply:
Although new patch adds `#[repr(u8)]`, should I remove it? Even if adding it, we cannot remove cast.